
1. Data Preprocessing

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.preprocessing import StandardScaler

df = pd.read_csv('BankCustomerData (1).csv')
df_dummies = pd.get_dummies(df, drop_first=True)

print(df.isnull().sum())

age             0
job             0
marital         0
education       0
default         0
balance         0
housing         0
loan            0
contact         0
day             0
month           0
duration        0
campaign        0
pdays           0
previous        0
poutcome        0
term_deposit    0
dtype: int64


In [ ]:
df_dummies['subscribed'] = (df['balance']>0).astype(int)

2. Feature Selection

In [ ]:
x = df_dummies.drop(['age','subscribed','duration','campaign','previous','subscribed'],axis=1)
y = df_dummies['subscribed']

3. Data Splitting

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size=0.2, random_state=80)

scaler = StandardScaler()
x_train_scaled=scaler.fit_transform(x_train)
x_test_scaled=scaler.fit_transform(x_test)

4. Model Training

In [ ]:
model = LogisticRegression()
model.fit(x_train_scaled,y_train)

LogisticRegression()

5. Model Evaluation

In [ ]:
y_pred = model.predict(x_test_scaled)

In [ ]:
accuracy = accuracy_score(y_test,y_pred)
conf_mat = confusion_matrix(y_test,y_pred)
class_report = classification_report(y_test,y_pred)

print(f"Accuracy: {accuracy}")
print(f"")
print(f"Confusion Matrix: ")
print(conf_mat)
print(f"")
print(f"Classification Report: ")
print(class_report)

Accuracy: 0.9371482176360225

Confusion Matrix: 
[[1430    3]
 [ 533 6562]]

Classification Report: 
              precision    recall  f1-score   support

           0       0.73      1.00      0.84      1433
           1       1.00      0.92      0.96      7095

    accuracy                           0.94      8528
   macro avg       0.86      0.96      0.90      8528
weighted avg       0.95      0.94      0.94      8528



6. Conclusion

The prediction model has given a 0.94 rounded (93.7%) accuracy value, this is a good range for the target variable to be within. Based on this result, it can be deduced that the bank's marketing strategy is very effective to achieve a success percentage of over 90%.